### PR TITLE
Minor side effect universe cleanups

### PR DESCRIPTION
**Pull request by SkySkimmer**
I think this works but not sure.

Depends on #10862

### CODE REVIEW

**Comment by ppedrot** [comment]
@SkySkimmer can you rebase?

File: test-suite/bugs/closed/bug_6323.v [hunk 1]
```diff
@@ -6,4 +6,5 @@ Goal True.
       simple refine (let id' : { x : X' | True } -> X' := _ in _);
       [ abstract refine (@proj1_sig _ _) | ]
     ].
-Abort.
+  exact I.
```
[the remark this answers is not in the document]
Comment (JasonGross) [reply test-suite/bugs/closed/bug_6323.v]: Ah, nevermind, you explain it in the commit message

**Comment by JasonGross** [comment]
Should this wait for merging until we can get #10911 working, or is it orthogonal?

**Comment by SkySkimmer** [comment]
This breaks rewriter and I was not confident about the semantics of the change from the start so let's give up on it. I'm going to leave just the minor cleanup commits in this PR.

**Review by ppedrot — APPROVED** [review]

**Comment by ppedrot** [comment]
Will merge soon if no objection.

**Pull request merged**